ApplicationInsights/JavaScriptSDK/ApplicationInsights.ts

```diff
@@ -199,7 +201,79 @@ export class ApplicationInsights implements IAppInsights, ITelemetryPlugin, IApp
         }
     }
 
+    /**
+     * Log an exception you have caught.
+     *
+     * @param {IExceptionTelemetry} exception   Object which contains exception to be sent
+     * @param {{[key: string]: any}} customProperties   Additional data used to filter pages and metrics in the portal. Defaults to empty.
+     *
+     * Any property of type double will be considered a measurement, and will be treated by Application Insights as a metric.
+     * @memberof ApplicationInsights
+     */
+    public trackException(exception: IExceptionTelemetry, customProperties?: {[key: string]: any}): void {
+        try {
+            let telemetryItem: ITelemetryItem = TelemetryItemCreator.createItem(
+                this._logger,
+                exception,
+                Exception.dataType,
+                Exception.envelopeType,
+                customProperties
+            );
+            this.core.track(telemetryItem);
+        } catch (e) {
+            this._logger.throwInternal(
+                LoggingSeverity.CRITICAL,
+                _InternalMessageId.TrackExceptionFailed,
+                "trackException failed, exception will not be collected: " + Util.getExceptionName(e),
+                { exception: Util.dump(e) });
+        }
+    }
+
+    /**
+     * @description Custom error handler for Application Insights Analytics
+     * @param {IAutoExceptionTelemetry} exception
+     * @memberof ApplicationInsights
+     */
+    public _onerror(exception: IAutoExceptionTelemetry): void {
```

Comment:
did this exist before?

Reply:
Yes. This is the method that will be called by the snippet (snippet.js line 53) in the case of auto collection. It essentially calls trackException. 

---
In reply to: [217542252](https://github.com/Microsoft/ApplicationInsights-JS/pull/668#discussion_r217542252) [](ancestors = 217542252)